tests/integration/framework/iowriter/iowriter.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iowriter
+
+import (
+	"bytes"
+	"io"
+	"testing"
+)
+
+type stdwriter struct {
+	t *testing.T
+}
+
+func New(t *testing.T) io.WriteCloser {
+	return &stdwriter{t}
+}
+
+func (w *stdwriter) Write(p []byte) (n int, err error) {
+	for _, line := range bytes.Split(p, []byte("\n")) {
```

Comment:
This may have issues if the data is flushed before the line is complete.

For example, assume the producer wants to log `supercalifragilisticexpialidocious` but because of internal buffering, it is sent to `Write()` in 2 separate calls:

- `supercalifragil`
- `isticexpialidocious`

This will log that in 2 separate lines

Better would be to:

- When data comes in, rather than splitting for `\n`, just iterate byte-by-byte, storing everything in an internal buffer.
- If you encounter `\n` or EOF, then print out whatever is in the internal buffer.
- At the end of the loop in the Write method you may have data left into the internal buffer
- That data should be flushed out in the Close method.

Reply:
Makes sense :+1: Thanks @ItalyPaleAle :slightly_smiling_face: